weldx/transformations.py

```diff
@@ -2423,11 +2401,14 @@ def get_cs(
         lcs_result = LocalCoordinateSystem()
         for edge in path_edges:
             lcs = self.graph.edges[edge]["lcs"]
-            if lcs.time is not None:
-                if lcs.reference_time is None and self.reference_time is not None:
-                    lcs = deepcopy(lcs)
+            if lcs.is_time_dependent:
+                if not lcs.has_reference_time and self.has_reference_time:
+                    time_lcs = time_interp + (time_ref_interp - self.reference_time)
+                    lcs = lcs.interp_time(time_lcs)
                     lcs.reset_reference_time(self.reference_time)
-                lcs = lcs.interp_time(time_interp, time_ref_interp)
+                    lcs.reset_reference_time(time_ref_interp)
+                else:
+                    lcs = lcs.interp_time(time_interp, time_ref_interp)
```

Comment:
I added this as an alternative to creating a deepcopy
Basically create a single `TimedeltaIndex` for LCS interpolation that is shifted to match the CSM reference time. Need to add time_ref data afterwards and shift to final `time_ref_interp`

we can also keep the old code @vhirtham

Reply:
As you like, but isn't `lcs.reset_reference_time(self.reference_time)` obsolete? Why do we need to reset the reference time twice directly after each other? The second call should be independent of the first one or am I missing something?